src/test/java/net/masterthought/cucumber/ReportBuilderTest.java

```diff
@@ -62,6 +62,27 @@ void ReportBuilder_storesFilesAndConfiguration() {
         assertThat(assignedConfiguration).isSameAs(configuration);
     }
 
+    @Test
+    void ReportBuilder_setsAndGetsCustomReportParser(){
+        // given
+        final List<String> jsonFiles = new ArrayList<>();
+        final Configuration configuration = new Configuration(null, null);
+        final ReportParser reportParser = new ReportParser(configuration);
+
+        // when
+        ReportBuilder builder = new ReportBuilder(jsonFiles, configuration);
+        builder.setReportParser(reportParser);
+
+        // then
+        List<String> assignedJsonReports = Whitebox.getInternalState(builder, "jsonFiles");
```

Comment:
no need to validate everything here - just assert what you are trying to validate ->  leave only line 8 in this section

by this this is clear that this unit test validates only one particular method while others are validated by different tests

Reply:
sure Damian! I will update this.